plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/ServicesAndRoutingPage.java

```diff
@@ -124,19 +126,28 @@ protected void doCreateControls(Composite parent, DataBindingContext dbc) {
         .to(BeanProperties.value(IServiceAndRoutingPageModel.PROPERTY_ADD_ROUTE)
         .observe(model))
         .in(dbc);
-        ValueBindingBuilder.bind(WidgetProperties.text(SWT.Modify).observe(textRouteHostname))
+        final IObservableValue<String> routeHostnameObservable = WidgetProperties.text(SWT.Modify).observe(textRouteHostname);
+        ValueBindingBuilder.bind(routeHostnameObservable)
         .converting(new TrimmingStringConverter())
-        .validatingAfterConvert(new IValidator() {
+        .to(BeanProperties.value(IServiceAndRoutingPageModel.PROPERTY_ROUTE_HOSTNAME)
+                .observe(model))
+        .in(dbc);
+        
+        MultiValidator validator = new MultiValidator() {
             
             @Override
-            public IStatus validate(Object value) {
-                boolean valid = DomainValidator.getInstance(true).isValid((String) value);               
-                return valid?ValidationStatus.ok():ValidationStatus.error(NLS.bind(OpenShiftUIMessages.InvalidHostNameErrorMessage, value));
+            protected IStatus validate() {
+                IStatus status = ValidationStatus.ok();
+                boolean isAddRoute = addRouteModelObservable.getValue();
+                String hostName = routeHostnameObservable.getValue();
+                if (isAddRoute && StringUtils.isNotBlank(hostName) && !DomainValidator.getInstance(true).isValid(hostName)) {
+                    status = ValidationStatus.error(NLS.bind(OpenShiftUIMessages.InvalidHostNameErrorMessage, hostName));
+                }
+                return status;
             }
-        })
-        .to(BeanProperties.value(IServiceAndRoutingPageModel.PROPERTY_ROUTE_HOSTNAME)
-        .observe(model))
-        .in(dbc);
+        };
+        dbc.addValidationStatusProvider(validator);
+        ControlDecorationSupport.create(validator, SWT.LEFT | SWT.TOP);
 
 	}
 
```

Comment:
and to mark CANCEL and ERROR as error states, the wizardPageSupport needs to be initialized as follows (otherwise CANCEL wont disable "Next >"):

```
	@Override
	protected void setupWizardPageSupport(DataBindingContext dbc) {
		ParametrizableWizardPageSupport.create(IStatus.ERROR | IStatus.CANCEL, this, dbc);
	}
```